### PR TITLE
DB-918 - Connector & Line Formatting

### PR DESCRIPTION
**Pull request by drewfulton**
This PR handles new formatting for lines and for connectors.

# Lines
For lines, we now offer a handful of different options, not all of which I am entirely sure what they do.  The main one we want is the head/tail formatting so we can add various arrow heads.  There's also some details about how lines join together and a few other properties that we don't need but I implemented while I was at it.

# Connectors
Connectors were a bit awkward.  I ended up doing a bit of a workaround rather than a complete rewrite of how Steve handles autoshapes.  I'll leave a couple comments in the code about this but ultimately it works.

### CODE REVIEW

File: pptx/oxml/shapes/autoshape.py [hunk 1]
```diff
@@ -238,8 +239,25 @@ class CT_PresetGeometry2D(BaseOxmlElement):
     """
 
     avLst = ZeroOrOne("a:avLst")
-    prst = RequiredAttribute("prst", MSO_AUTO_SHAPE_TYPE)
+    preset = RequiredAttribute("prst", ST_ShapeType)
 
+    @property
```
Comment (drewfulton) [inline pptx/oxml/shapes/autoshape.py:244]: This was the workaround I mentioned in the description.  The way Steve originally set this up, he is using AutoShapeTypes as the list of possibilities for `prst` and throughout the code, it is expecting a member of that ENUM.  Unfortunately, Connectors are not part of that list.  Both Connectors and AutoShapes inherit from BaseShape but aren't the same.  As a result, I found this little workaround that seems to let us use AutoShapes as usual but also use connectors when we want.

**Comment by drewfulton** [comment]
I'll come back and take a look at the failing tests first thing tomorrow.

**Comment by drewfulton** [comment]
Circle CI is now fixed and the few broken unit tests are also fixed.

File: pptx/dml/line.py [hunk 2]
```diff
@@ -165,3 +251,162 @@ def width(self):
...
+    @property
+    def limit(self):
+        """
+        Property defining the angle of the miter.  Must be a positive percentage
+        """
+        return self._element.lim
+
+    @limit.setter
+    def limit(self, val):
+        self._element.lim = val
```
Comment (luna215) [inline pptx/dml/line.py:373]: Based on the doc string on line 367 limit shouldn't be negative, so I'm thinking that we should add a check for negative values, or are we not setting this from the client side?

**Review by luna215 — CHANGES_REQUESTED** [review]
Nicely done! I left a small request change

**Comment by drewfulton** [comment]
That is handled by the validation built in to the simple type

File: pptx/dml/line.py (hunk 2 above)
Comment (drewfulton) [reply pptx/dml/line.py:373]: This happens automatically by the validation as part of the simple type class

File: pptx/oxml/simpletypes.py [hunk 3]
```diff
@@ -587,6 +587,21 @@ class ST_PositiveFixedPercentage(ST_Percentage):
     def validate(cls, value):
         cls.validate_float_in_range(value, 0.0, 1.0)
 
+class ST_PositivePercentage(ST_Percentage):
```
Comment (drewfulton) [inline pptx/oxml/simpletypes.py:590]: This is the simpletype class that does the validation for us.

File: pptx/dml/line.py (hunk 2 above)
Comment (drewfulton) [reply pptx/dml/line.py:373]: I left another comment where this is happening

Comment (luna215) [reply pptx/dml/line.py:373]: Thank you!

**Review by luna215 — COMMENTED** [review]
💯

**Review by luna215 — APPROVED** [review]

**Comment by drewfulton** [comment]
@luna215 , just had to make another update before release.

**Review by luna215 — APPROVED** [review]

**Pull request merged**